dspace-api/src/test/java/org/dspace/statistics/export/processor/ExportEventProcessorTest.java

```diff
@@ -0,0 +1,325 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.statistics.export.processor;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.UnsupportedEncodingException;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import javax.servlet.http.HttpServletRequest;
+
+import org.dspace.AbstractDSpaceTest;
+import org.dspace.content.Entity;
+import org.dspace.content.EntityType;
+import org.dspace.content.Item;
+import org.dspace.content.MetadataValue;
+import org.dspace.content.service.EntityService;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.dspace.services.ConfigurationService;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/**
+ * Test for the ExportEventProcessor class
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class ExportEventProcessorTest extends AbstractDSpaceTest {
+
+    @Mock
+    private Context context = mock(Context.class);
+    @Mock
+    private HttpServletRequest request = mock(HttpServletRequest.class);
+    @Mock
+    private Item item = mock(Item.class);
+    @Mock
+    private ConfigurationService configurationService = mock(ConfigurationService.class);
+    @Mock
+    private ItemService itemService = mock(ItemService.class);
+    @Mock
+    private EntityService entityService = mock(EntityService.class);
+
+
+    @InjectMocks
+    ExportEventProcessor exportEventProcessor = mock(ExportEventProcessor.class);
+
+
+    @Test
+    /**
+     * Test the getBaseParameters method
+     */
+    public void testGetBaseParameters() throws UnsupportedEncodingException {
+        exportEventProcessor.context = context;
+        exportEventProcessor.request = request;
+        exportEventProcessor.configurationService = configurationService;
+        exportEventProcessor.trackerUrlVersion = "Z39.88-2004";
+
+        when(exportEventProcessor.getCurrentDateString()).thenReturn("2020-01-24T13:24:33Z");
+
+        when(request.getRemoteAddr()).thenReturn("test-client-ip");
+        when(request.getHeader("USER-AGENT")).thenReturn("test-user-agent");
+        when(request.getHeader("referer")).thenReturn("test-referer");
+        when(configurationService.getBooleanProperty("useProxies", false)).thenReturn(false);
+        when(configurationService.getProperty("dspace.hostname")).thenReturn("localhost");
```

Comment:
`dspace.hostname` is not valid.  Again here, you should not need to mock configurations. Just set them to the values you need.

Reply:
Updated the config parameter